_posts/2020-02-17-Combining-MongoDB-and-AWS-Lambda.md

```diff
@@ -0,0 +1,364 @@
+---
+layout: post
+authors: [nick_van_hoof]
+title: 'Marrying MongoDB Atlas and AWS Lambda'
+image: /img/2020-02-17-Combining-MongoDB-Atlas-and-AWS-Lambda/featured-image.png
+tags: [MongoDB, Cloud, AWS, AWS Lambda, Serverless]
+category: Cloud
+comments: false
+---
+
+### Reading time: 10 min 30 sec
+
+# Table of contents
+* [MongoDB and AWS Lambda: a successful marriage?](#mongodb-and-aws-lambda-a-successful-marriage)
+* [Why?](#mongodb-and-aws-lambda-why)
+* [Performance](#mongodb-and-aws-lambda-performance)
+* [Cold start vs warm performance](#cold-start-vs-warm-performance)
+* [Performance conclusions](#performance-conclusion)
+* [VPC peering between AWS and MongoDB Atlas](#vpc-peering-connect-your-lambda-functions-with-your-mongodb-atlas-cluster)
+* [Useful links](#useful-links)
+
+
+# MongoDB and AWS Lambda: a successful marriage?
+
+<div style="text-align: center;" >
+  <img src="/img/2020-02-17-Combining-MongoDB-Atlas-and-AWS-Lambda/rings.png" width="15%" height="15%">
+</div>
+
+Can we use MongoDB Atlas when working with AWS Lambda?  
+Yes we can!   
+It's simple enough to setup and above all also performing well!  
+
+This blog concerns mainly two things:
+* Why would you use MongoDB + AWS Lambda and how does it perform
+* How to create a production grade setup with vpc-peering
+
+# Why?
+As I am both a fan of AWS Lambda and MongoDB Atlas it was fun for me to marry them.  
+However in the real world we don't do things for fun alone.  
+
+<div style="text-align: center;" >
+  <img src="/img/2020-02-17-Combining-MongoDB-Atlas-and-AWS-Lambda/mongodb-plus-aws-lambda.png" width="70%" height="70%">
+</div>
+
+What are the motives to combine MongoDB Atlas and AWS Lambda?
+
+* Combine the serverless capabilities of Lambda with the MongoDB strong points
+* Payment model - In case of MongoDB you provision your cluster and you know what you'll pay for it, clusters can grow with your business without downtime or code changes.
+* Flexible Data Acces - MongoDB has a rich query language and aggregation framework. On top of your data in MongoDB you can build nice dashboards for business intelligence. (eg. [MongoDB Charts](https://www.mongodb.com/products/charts))
+* Indexes - Supports up to 64 indexes per collection with a wide variety of index types like hash, compound, unique, array, partial, TTL, geospatial, sparse, text and wildcard indexes
+* Large documents allowed. A MongoDB document can be up to 16 Mb. 
+* Performance - a built-in cache and support for lots of secondary indexes that can span across arrays and subdocuments, making virtually all queries very fast
+* Tunable consistency[^1] 
+* Observability - MongoDB exposes more than 100 different metrics and has a built-in performance advisor. Because "you can't optimize what you can't measure."
+* Platform capabilities[^2] 
+* Joinable documents [^3] 
+
+# Performance
```

Comment:
What is Performance in this context?
Explain you are measuring the time it takes to store data in the database, with a "cold" and "warm" start

Reply:
I made updates. You can validate them again.